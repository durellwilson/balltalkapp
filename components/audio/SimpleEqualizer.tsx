import React, { useState } from 'react';
import { View, Text, StyleSheet, Platform } from 'react-native';
import Slider from '@react-native-community/slider';

interface EqualizerBand {
  frequency: number;
  gain: number;
}

export default function SimpleEqualizer() {
  const [bands, setBands] = useState<EqualizerBand[]>([
    { frequency: 60, gain: 0 },
    { frequency: 230, gain: 0 },
    { frequency: 910, gain: 0 },
    { frequency: 3600, gain: 0 },
    { frequency: 14000, gain: 0 }
  ]);
  
  const updateBand = (index: number, gain: number) => {
    const newBands = [...bands];
    newBands[index].gain = gain;
    setBands(newBands);
    
    // Log the updated equalizer settings
    console.log('Equalizer settings updated:', newBands);
    
    // In a real implementation, this would update audio processing
    // For example: audioProcessor.setEqualizer(newBands);
  };
  
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Simple Equalizer</Text>
      <Text style={styles.subtitle}>Adjust frequency bands to shape your sound</Text>
      
      {bands.map((band, index) => (
        <View key={index} style={styles.bandContainer}>
          <Text style={styles.frequencyText}>{band.frequency} Hz</Text>
          <Slider
            style={styles.slider}
            minimumValue={-12}
            maximumValue={12}
            step={0.1}
            value={band.gain}
            minimumTrackTintColor="#007AFF"
            maximumTrackTintColor="#DDDDDD"
            onValueChange={(value) => updateBand(index, value)}
          />
          <Text style={styles.gainText}>{band.gain.toFixed(1)} dB</Text>
        </View>
      ))}
      
      <View style={styles.infoContainer}>
        <Text style={styles.infoText}>Platform: {Platform.OS}</Text>
        <Text style={styles.infoText}>
          Note: This is a UI demonstration. In a real implementation, these controls would be connected 
          to audio processing libraries.
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    padding: 20,
    backgroundColor: '#f5f5f5',
    borderRadius: 8,
    margin: 16,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 8,
    color: '#333',
  },
  subtitle: {
    fontSize: 14,
    marginBottom: 20,
    color: '#666',
  },
  bandContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 16,
  },
  frequencyText: {
    width: 80,
    fontSize: 14,
    color: '#333',
  },
  slider: {
    flex: 1,
    height: 40,
    marginHorizontal: 10,
  },
  gainText: {
    width: 60,
    fontSize: 14,
    color: '#333',
    textAlign: 'right',
  },
  infoContainer: {
    marginTop: 20,
    padding: 12,
    backgroundColor: '#e9e9e9',
    borderRadius: 6,
  },
  infoText: {
    fontSize: 12,
    color: '#666',
    marginBottom: 4,
  },
}); 