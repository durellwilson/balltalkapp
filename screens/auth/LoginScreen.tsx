import React, { useState } from 'react';
import { View, TextInput, Button, Text, StyleSheet, ActivityIndicator, Platform } from 'react-native';
import { auth } from '../../config/firebase';
import { signInWithEmailAndPassword } from 'firebase/auth';

export default function LoginScreen() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');
  const [loading, setLoading] = useState(false);
  const [success, setSuccess] = useState(false);
  
  async function handleLogin() {
    if (!email || !password) {
      setError('Email and password are required');
      return;
    }
    
    setLoading(true);
    setError('');
    setSuccess(false);
    
    try {
      const userCredential = await signInWithEmailAndPassword(auth, email, password);
      console.log('User logged in successfully:', userCredential.user.uid);
      setSuccess(true);
      // In a real app, navigation would be handled by an auth state listener
      // navigation.navigate('Home');
    } catch (err) {
      console.error('Login error:', err);
      setError(err instanceof Error ? err.message : String(err));
      setSuccess(false);
    } finally {
      setLoading(false);
    }
  }
  
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Login to BallTalk</Text>
      
      <View style={styles.formContainer}>
        <TextInput
          style={styles.input}
          placeholder="Email"
          value={email}
          onChangeText={setEmail}
          autoCapitalize="none"
          keyboardType="email-address"
          editable={!loading}
        />
        
        <TextInput
          style={styles.input}
          placeholder="Password"
          value={password}
          onChangeText={setPassword}
          secureTextEntry
          editable={!loading}
        />
        
        {error ? <Text style={styles.error}>{error}</Text> : null}
        {success ? <Text style={styles.success}>Login successful!</Text> : null}
        
        {loading ? (
          <ActivityIndicator size="large" color="#007AFF" style={styles.loader} />
        ) : (
          <Button
            title="Login"
            onPress={handleLogin}
            disabled={loading}
          />
        )}
      </View>
      
      <View style={styles.infoContainer}>
        <Text style={styles.infoText}>Platform: {Platform.OS}</Text>
        <Text style={styles.infoText}>Firebase Auth initialized: {auth ? '✅' : '❌'}</Text>
        <Text style={styles.infoText}>
          Note: This is a test login screen. In a real app, you would have proper navigation
          and user state management.
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: '#f5f5f5',
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 20,
    color: '#333',
    textAlign: 'center',
    marginTop: 40,
  },
  formContainer: {
    backgroundColor: 'white',
    borderRadius: 10,
    padding: 20,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  input: {
    borderWidth: 1,
    borderColor: '#ddd',
    padding: 12,
    marginBottom: 16,
    borderRadius: 6,
    fontSize: 16,
  },
  error: {
    color: '#d32f2f',
    marginBottom: 16,
    fontSize: 14,
  },
  success: {
    color: '#388e3c',
    marginBottom: 16,
    fontSize: 14,
  },
  loader: {
    marginVertical: 16,
  },
  infoContainer: {
    marginTop: 30,
    padding: 16,
    backgroundColor: '#e9e9e9',
    borderRadius: 8,
  },
  infoText: {
    fontSize: 14,
    color: '#666',
    marginBottom: 6,
  },
}); 