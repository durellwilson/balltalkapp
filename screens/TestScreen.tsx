import React from 'react';
import { ScrollView, Text, StyleSheet, View, Platform } from 'react-native';
import FirebaseVerification from '../components/FirebaseVerification';
import BasicRecorder from '../components/audio/recorder/BasicRecorder';
import SimpleEqualizer from '../components/audio/SimpleEqualizer';

export default function TestScreen() {
  return (
    <ScrollView style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.heading}>BallTalk Testing</Text>
        <Text style={styles.subheading}>Platform: {Platform.OS}</Text>
      </View>
      
      <View style={styles.section}>
        <Text style={styles.sectionTitle}>Firebase Connectivity</Text>
        <FirebaseVerification />
      </View>
      
      <View style={styles.section}>
        <Text style={styles.sectionTitle}>Audio Recording</Text>
        <BasicRecorder />
      </View>
      
      <View style={styles.section}>
        <Text style={styles.sectionTitle}>Audio Processing</Text>
        <SimpleEqualizer />
      </View>
      
      <View style={styles.footer}>
        <Text style={styles.footerText}>
          BallTalk App - Development Version
        </Text>
      </View>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f0f0f0',
  },
  header: {
    padding: 20,
    backgroundColor: '#007AFF',
    paddingTop: Platform.OS === 'ios' ? 60 : 40,
  },
  heading: {
    fontSize: 28,
    fontWeight: 'bold',
    color: 'white',
    marginBottom: 8,
  },
  subheading: {
    fontSize: 16,
    color: 'rgba(255, 255, 255, 0.8)',
  },
  section: {
    marginVertical: 10,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginHorizontal: 16,
    marginTop: 16,
    marginBottom: 8,
    color: '#333',
  },
  footer: {
    padding: 20,
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 40,
  },
  footerText: {
    fontSize: 14,
    color: '#666',
  },
}); 