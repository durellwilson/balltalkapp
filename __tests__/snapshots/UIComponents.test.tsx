import React from 'react';
import renderer from 'react-test-renderer';
import SimpleEqualizer from '../../components/audio/SimpleEqualizer';
import LoginScreen from '../../screens/auth/LoginScreen';
import TestScreen from '../../screens/TestScreen';

// Mock components for TestScreen
jest.mock('../../components/FirebaseVerification', () => 'MockFirebaseVerification');
jest.mock('../../components/audio/recorder/BasicRecorder', () => 'MockBasicRecorder');
jest.mock('../../components/audio/SimpleEqualizer', () => 'MockSimpleEqualizer');

// Mock the Slider for SimpleEqualizer
jest.mock('@react-native-community/slider', () => 'MockSlider');

// Mock Firebase Auth
jest.mock('firebase/auth', () => ({
  signInWithEmailAndPassword: jest.fn()
}));

// Mock Firebase config
jest.mock('../../config/firebase', () => ({
  auth: {},
  firebaseDb: {}
}));

describe('UI Component Snapshots', () => {
  it('SimpleEqualizer matches snapshot', () => {
    // Reset the mock to use the real component for this test
    jest.unmock('../../components/audio/SimpleEqualizer');
    
    const tree = renderer.create(<SimpleEqualizer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('LoginScreen matches snapshot', () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('TestScreen matches snapshot', () => {
    const tree = renderer.create(<TestScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 