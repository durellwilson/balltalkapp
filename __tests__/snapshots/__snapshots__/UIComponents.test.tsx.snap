// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component Snapshots LoginScreen matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
        "marginTop": 40,
        "textAlign": "center",
      }
    }
  >
    Login to BallTalk
  </Text>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 3,
        "padding": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      editable={true}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        {
          "borderColor": "#ddd",
          "borderRadius": 6,
          "borderWidth": 1,
          "fontSize": 16,
          "marginBottom": 16,
          "padding": 12,
        }
      }
      value=""
    />
    <TextInput
      editable={true}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        {
          "borderColor": "#ddd",
          "borderRadius": 6,
          "borderWidth": 1,
          "fontSize": 16,
          "marginBottom": 16,
          "padding": 12,
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#e9e9e9",
        "borderRadius": 8,
        "marginTop": 30,
        "padding": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 6,
        }
      }
    >
      Platform: 
      ios
    </Text>
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 6,
        }
      }
    >
      Firebase Auth initialized: 
      âœ…
    </Text>
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 6,
        }
      }
    >
      Note: This is a test login screen. In a real app, you would have proper navigation and user state management.
    </Text>
  </View>
</View>
`;

exports[`UI Component Snapshots SimpleEqualizer matches snapshot 1`] = `<MockSimpleEqualizer />`;

exports[`UI Component Snapshots TestScreen matches snapshot 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#f0f0f0",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "#007AFF",
          "padding": 20,
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 28,
            "fontWeight": "bold",
            "marginBottom": 8,
          }
        }
      >
        BallTalk Testing
      </Text>
      <Text
        style={
          {
            "color": "rgba(255, 255, 255, 0.8)",
            "fontSize": 16,
          }
        }
      >
        Platform: 
        ios
      </Text>
    </View>
    <View
      style={
        {
          "marginVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 8,
            "marginHorizontal": 16,
            "marginTop": 16,
          }
        }
      >
        Firebase Connectivity
      </Text>
      <MockFirebaseVerification />
    </View>
    <View
      style={
        {
          "marginVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 8,
            "marginHorizontal": 16,
            "marginTop": 16,
          }
        }
      >
        Audio Recording
      </Text>
      <MockBasicRecorder />
    </View>
    <View
      style={
        {
          "marginVertical": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 8,
            "marginHorizontal": 16,
            "marginTop": 16,
          }
        }
      >
        Audio Processing
      </Text>
      <MockSimpleEqualizer />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 40,
          "marginTop": 20,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 14,
          }
        }
      >
        BallTalk App - Development Version
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
